### ISSUE TITLE
No assets expiring even though I have assets which have expired or are close to expiring

### ISSUE THREAD

**Issue by bradbakerdx** [snipe/snipe-it#1760]
Under assets -> asset maintenance I have multiple assets which are either expired or close to expiring. For example I have this asset:

DB1 Dell    Maintenance 4 Hour 7x24 Onsite  2012-11-02  2015-11-02  1095
Server04    Dell    Maintenance 4 Hour 7x24 Onsite  2011-03-17  2016-03-17  1827

Its expiring in just over a month yet when I run  /usr/bin/php /var/www/asset-mgr/artisan alerts:expiring

I get: 

0 expiring assets
0 expiring licenses

Is there a way to configure what is considered expiring?

Thanks,
Brad

**Comment by bradbakerdx** [comment]
Looking at /app/commands/SendExpirationAlerts.php I see that its getting assets expiring in the next 60 days:

`$expiring_assets = Asset::getExpiringWarrantee(60);`

Looking at /app/models/Asset.php I see that its determining the expiring warranty using the following code:

  ```
public static function getExpiringWarrantee( $days = 30 )
  {

      return Asset::where( 'archived', '=', '0' )
                  ->whereNotNull( 'warranty_months' )
                  ->whereNotNull( 'purchase_date' )
                  ->whereNull( 'deleted_at' )
                  ->whereRaw( DB::raw( 'DATE_ADD(`purchase_date`,INTERVAL `warranty_months` MONTH) <= DATE(NOW() + INTERVAL '
                                       . $days
                                       . ' DAY) AND DATE_ADD(`purchase_date`,INTERVAL `warranty_months` MONTH) > NOW()' ) )
                  ->orderBy( 'purchase_date', 'ASC' )
                  ->get();

  }
```

So it appears the purchase date MUST be set - in our case we have many assets purchased before I even came on board and there's no easy way of determining the purchase date - so in many cases its not set. :-(  

I'd be nice if there was some sort of indicator that must be set for expiration alerts to work. Also it looks like warranty_months must be set - again looking at our database directly it appears that value is null for all our assets.  

I can certainly set the purchase_date value to something - even if I need to make it up. However I'm not sure why warranty_months is null for all our assets?

**Comment by bradbakerdx** [comment]
I figured out how to set the purchase date for all our assets, as well as the warranty month but I'm still confused as to how this is meant to work. 

I have my asset purchase date set to 2011-03-17, the original purchase warranty is 12 months. I thought that I needed to add asset maintenance with the "warranty improvement" flag set for subsequent warranty renewals.  So I created two different asset maintenance entries:

ProSupport	        3/17/2011	3/17/2016
4 Hour 7x24 Onsite	3/18/2012	3/17/2016

Then I ran the asset 
/usr/bin/php /var/www/asset-mgr/artisan alerts:expiring

And I was expecting to get an alert that one server was coming up for renewal in 30 days but it doesn't. Instead I get:

0 expiring assets
0 expiring licenses

Looking at the code and substituting values I get that its looking for systems matching:
2011-03-17 (purchase date) + 12 months (warranty months)  <= 2016-02-22 (today's date) + 30  (true)  
AND 
2011-03-17 (purchase date) + 12 Months > 2016-02-22 (today's date) (false)

This doesn't even seem to be taking into account asset maintenance at all? Is this a bug or is this feature just not complete/functional? 

Seems like for this to work we'd have to put in our original purchase date + total of all asset maintenance in months. That's awkward/tedious to maintain? 

I don't understand what the purpose of asset maintenance is and what warranty improvement is meant to do?
